apps/snitch_core/test/support/factory/factory.ex

```diff
@@ -201,4 +211,17 @@ defmodule Snitch.Factory do
     count = Map.get(context, :tax_category_count, 3)
     [tax_categories: insert_list(count, :tax_category)]
   end
+
+  def tax_rate(_context) do
```

Comment:
You can emulate `tax_rate/1` by using `tax_rates/1`. `tax_rate/1` does not use it's argument.
**Ideally, you should pick the `zone` from `context` so that you minimise zone and tax category insertions in the test setup.**

_Moreover, `def tax_rate_factory` injects a `tax_rate/0` anyways._